transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -96,13 +98,11 @@ public ChannelPipeline addFirst(String name, ChannelHandler handler) {
 
     @Override
     public synchronized ChannelPipeline addFirst(EventExecutorGroup group, String name, ChannelHandler handler) {
-        name = filterName(name, handler);
-        addFirst0(new DefaultChannelHandlerContext(this, findInvoker(group), name, handler));
-        return this;
+        return addFirst(findInvoker(group), name, handler);
```

Comment:
why this change? won't we now have to go through 2 `synchronized(this)` blocks when this method is called?